lib/libzrepl/data_conn.c

```diff
@@ -732,14 +748,18 @@ uzfs_zvol_rebuild_dw_replica(void *arg)
 				zinfo->quiesce_requested = 1;
 			}
 			mutex_exit(&zinfo->main_zv->rebuild_mtx);
-			/*
-			 * Wait for all outstanding IOs to be flushed
-			 * to disk before making further progress
-			 */
-			quiesce_wait(zinfo, 0);
 
 			if (start_rebuild_from_clone == 1) {
 				start_rebuild_from_clone = 2;
+				wait_snap_timeout();
+				/*
+				 * Wait for all outstanding IOs to be flushed
+				 * to disk before making further progress
+				 */
+				quiesce_wait(zinfo, 0);
```

Comment:
this need to be moved to after changing the state. Also, better to set the variables `quiesce_done` and `quiesce_requested` after setting the state.

Reply:
ahhhhh!!! damn. pasted at the wrong location :(